src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,122 @@
+            statement.setBigDecimal(2, book.getPrice());
+
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new RuntimeException(
+                        "Expected to insert at less one row, but inserted 0 rows");
+            }
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                book.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't add a new book, " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String sql = "SELECT * FROM books WHERE id = ?";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+            statement.setBigDecimal(1, BigDecimal.valueOf(id));
+            ResultSet resultSet = statement.executeQuery();
+
+            if (resultSet.next()) {
+                return Optional.of(getBookFromResulSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    "Cannot create a connection for getting by id from DB", e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        String sql = "SELECT * FROM books";
+        List<Book> books = new ArrayList<>();
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+            ResultSet resultSet = statement.executeQuery();
+
+            while (resultSet.next()) {
+                books.add(getBookFromResulSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    "Cannot create a connection for getting all from DB", e);
+        }
+        return books;
+    }
+
+    @Override
+    public Book update(Book book) {
+        String sql = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+            statement.setLong(3, book.getId());
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new RuntimeException("Expected to update one row, but updated 0 rows");
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot create a connection for updating DB data", e);
+        }
+        return findById(book.getId()).orElseThrow(
```

Comment:
Is session still open? 
```suggestion
        return book;
```